ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -128,6 +135,63 @@ message GetFlatTransactionResponse {
   Transaction transaction = 1;
 }
 
+message GetEventsByContractIdRequest {
+  // The contract id being queried
+  // Required
+  string contract_id = 1;
+
+  // The parties whose events the client expects to see.
+  // Events that are not visible for the parties in this collection will not be present in the response.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  repeated string requesting_parties = 2;
+
+}
+message GetEventsByContractIdResponse {
+  // The collection of events, will only contain CreateEvent and consuming ExerciseEvents
```

Comment:
add something like "on which one of the requesting parties is an informee".

I'd also say either here on in the request docs that as long as the contractId queried is a valid identifier, you get a response. If it never existed, has been pruned away or no event has one of the requesting parties as an informee, you get an empty array.

This is important for privacy reasons so we should make it part of the documented specs.